language/move-prover/docgen/src/docgen.rs

```diff
@@ -664,7 +691,179 @@ impl<'env> Docgen<'env> {
         }
     }
 
-    /// Gen header for TOC, returning label where we can later insert the content after
+    /// Generate a static call diagram (.svg) starting from the given function.
+    fn gen_call_diagram(&self, fun_id: QualifiedId<FunId>) {
+        let fun_env = self.env.get_function(fun_id);
+        let name_of = |env: &FunctionEnv| {
+            if fun_env.module_env.get_id() == env.module_env.get_id() {
+                env.get_simple_name_string()
+            } else {
+                Rc::from(format!("\"{}\"", env.get_name_string()))
+            }
+        };
+
+        let mut dot_src: String = String::from("digraph G {\n");
```

Comment:
Please correct me if I'm wrong, but it looks like all of the strings being pushed on to `dot_src` are newline-suffixed. What would you think of having `dot_src` be a `Vec<String>` and then when `gen_svg_file` is called passing `dot_src.join("\n")`?

Reply:
Done.